services/QuillLMS/spec/workers/enqueue_concept_results_migration_worker_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe EnqueueConceptResultsMigrationWorker, type: :worker do
+
+  context '#perform' do
+    # Weird gotcha here: if you create a new raw concept_result the factory
+    # calls the base activity_session factory.  That factory creates a NEW
+    # concept_result related to it, so `create(:concept_result)` secretly
+    # creates two different ConceptResult records.  By pre-specifying an
+    # ActivitySession with no concept_results, we can guarantee that the
+    # ConceptResult factory calls only generate one record
+    let!(:activity_session) { create(:activity_session_without_concept_results) }
+    let!(:concept_result) { create(:concept_result, activity_session: activity_session) }
+
+    it 'should find all ConceptResult records in batches' do
+      start = nil
+      finish = nil
+      batch_size = 1
+      stub_const("#{described_class}::BATCH_SIZE", batch_size)
+      select_double = double
+
+      expect(ConceptResult).to receive(:select).and_return(select_double)
+      expect(select_double).to receive(:find_in_batches).with(start: start, finish: finish, batch_size: batch_size).and_return([concept_result.id])
+
+       subject.perform(start, finish)
+    end
+
+    it 'should enqueue a CopyConceptResultsToStudentResponsesWorker' do
+      concept_result_ids = ConceptResult.select(:id).all.map(&:id)
+
+      expect(CopyConceptResultsToStudentResponsesWorker).to receive(:perform_async).with(concept_result_ids)
+
+      subject.perform(nil, nil)
+    end
+
+    it 'should take start params to skip existing ConceptResults' do
```

Comment:
Nit: These two specs would provide more test coverage if they actually filtered out ConceptResults via the values of `start` and `finish`.

Reply:
So, secretly they both are because the `context` block includes a `let!` that creates a `concept_result`.  That means that there are two in the DB, and the test conditions check to ensure only one is getting processed, but that's definitely not obvious when I look at them again.  I think it's reasonable to tweak the tests to make it more obvious that they are testing the conditions that they are to avoid confusion.